loadgen/kitchen_sink_executor_test.go

```diff
@@ -0,0 +1,415 @@
+	"go.temporal.io/api/common/v1"
+	"go.temporal.io/api/enums/v1"
+	"go.temporal.io/api/history/v1"
+	"go.temporal.io/api/workflowservice/v1"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/testsuite"
+	"go.uber.org/zap/zaptest"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+var (
+	sdks = []cmdoptions.Language{
+		cmdoptions.LangGo,
+		cmdoptions.LangJava,
+		cmdoptions.LangPython,
+		cmdoptions.LangTypeScript,
+		cmdoptions.LangDotNet,
+	}
+	onlySDK = os.Getenv("SDK")
+
+	testDir, _ = os.Getwd()
+	repoDir    = filepath.Dir(testDir)
+
+	testStart             = time.Now().Unix()
+	testTimeout           = 30 * time.Second
+	workerBuildTimeout    = 1 * time.Minute
+	workerShutdownTimeout = 5 * time.Second
+
+	workerMutex     sync.RWMutex
+	workerBuildOnce = map[cmdoptions.Language]*sync.Once{
+		cmdoptions.LangGo:         new(sync.Once),
+		cmdoptions.LangJava:       new(sync.Once),
+		cmdoptions.LangPython:     new(sync.Once),
+		cmdoptions.LangTypeScript: new(sync.Once),
+		cmdoptions.LangDotNet:     new(sync.Once),
+	}
+	workerBuildErrs    = map[cmdoptions.Language]error{}
+	workerCleanupFuncs []func()
+)
+
+type testCase struct {
+	name                 string
+	testInput            *TestInput
+	expectedHistory      string
+	unsupportedSDKs      []cmdoptions.Language
+	expectedErrorMessage string
+}
+
+// TestKitchensink tests specific kitchensink features across SDKs.
+// Use the `SDK` environment variable to run only a specific SDK.
+func TestKitchensink(t *testing.T) {
+	if os.Getenv("CI") != "" && onlySDK == "" {
+		t.Skip("Skipping kitchensink test in CI without specific SDK set")
+	}
+
+	t.Cleanup(func() {
+		workerMutex.Lock()
+		defer workerMutex.Unlock()
+		for _, cleanupFunc := range workerCleanupFuncs {
+			cleanupFunc()
+		}
+	})
+
+	// Start dev server.
+	devServer := startDevServer(t)
+	defer devServer.Stop()
+	devServerAddr := devServer.FrontendHostPort()
+
+	// Create Temporal client.
+	temporalClient := createTemporalClient(t, devServerAddr)
+	defer temporalClient.Close()
+
+	for _, tc := range []testCase{
+		{
+			name: "TimerAction",
+			testInput: &TestInput{
+				WorkflowInput: &WorkflowInput{
+					InitialActions: []*ActionSet{
+						{
+							Actions: []*Action{
+								{
+									Variant: &Action_Timer{
+										Timer: &TimerAction{
+											Milliseconds: 1,
+											AwaitableChoice: &AwaitableChoice{
+												Condition: &AwaitableChoice_WaitFinish{
+													WaitFinish: &emptypb.Empty{},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedHistory: `
+				WorkflowExecutionStarted
+				WorkflowTaskScheduled
+				WorkflowTaskStarted
+				WorkflowTaskCompleted
+				TimerStarted {"startToFireTimeout":"0.001s"}
+				TimerFired
+				WorkflowTaskScheduled
+				WorkflowTaskStarted
+				WorkflowTaskCompleted
+				WorkflowExecutionCompleted`,
```

Comment:
Server has `printHistory` helper func to print history in this format. You might want it here too.